pypdf/_writer.py

```diff
@@ -852,6 +852,11 @@ def clone_reader_document_root(self, reader: PdfReader) -> None:
         self._root = self._root_object.indirect_reference  # type: ignore[assignment]
         self._pages = self._root_object.raw_get("/Pages")
         self._flatten()
+        for p in self.flattened_pages:
+            o = self._objects[p.idnum - 1].get_object()
+            self._objects[p.idnum - 1] = PageObject(self, p)
+            for k, v in o.items():
+                self._objects[p.idnum - 1][k] = v
```

Comment:
Does this loop add all elements of the page to the writer as indirect objects? Why is this necessary?

Reply:
reviewing the PR I'v slighty simplified the code.